macros/date_spine.sql

```diff
@@ -0,0 +1,46 @@
+{% macro date_spine(table, datepart, start_date, end_date) %}
+
+/*
+call as follows:
+
+date_spine(
+    ref('organizations'),
+    "minute",
+    "to_date('01/01/2016', 'mm/dd/yyyy')",
+    "dateadd(week, 1, current_date)"
+)
+
+*/
+
+with rawdata as (
+
+    select * from {{ table }}
+
+),
+
+all_periods as (
+
+    select (
+        {{
+            dateadd(
+                datepart,
+                "row_number() over () - 1",
+                start_date
+            )
+        }}
+    ) as date_{{datepart}}
```

Comment:
I think this is clever, but I'm not sure how I feel about returning different fields based on the supplied arguments. On the one hand, the resulting field names will be explicit. On the other, this could break downstream models if the datepart is changed from `day` to `week`, for instance. Maybe that's a good thing?